toolchain/check/handle_binding_pattern.cpp

```diff
@@ -64,6 +64,13 @@ static auto HandleAnyBindingPattern(Context& context, Parse::NodeId node_id,
   auto push_bind_name = [&](SemIR::InstId bind_id) {
     context.node_stack().Push(node_id, bind_id);
     if (is_generic && !is_associated_constant) {
+      // TODO: this is a temporary workaround until bind_id has a more
+      // consistent kind.
+      if (auto binding_pattern =
+              context.insts().TryGetAs<SemIR::BindingPattern>(bind_id);
+          binding_pattern.has_value()) {
```

Comment:
```suggestion
              context.insts().TryGetAs<SemIR::BindingPattern>(bind_id)) {
```

Does this work as an implicit boolean test? I think that's what we're generally doing with things that return an optional.